### ISSUE TITLE
regression: MIR assertion failure ICE

### ISSUE THREAD

**Issue by Mark-Simulacrum** [rust-lang/rust#94506]
[Crater](https://github.com/rust-lang/rust/issues/94345) picked up this issue:

* https://crater-reports.s3.amazonaws.com/beta-1.60-1/beta-2022-02-22/gh/playXE.Jazz/log.txt
* https://crater-reports.s3.amazonaws.com/beta-1.60-1/beta-2022-02-22/gh/Ahrimanox.strategy_game/log.txt
* https://crater-reports.s3.amazonaws.com/beta-1.60-1/beta-2022-02-22/gh/Sonnenhut.aoc2019/log.txt
* https://crater-reports.s3.amazonaws.com/beta-1.60-1/beta-2022-02-22/gh/aaronjeline.rvm/log.txt
* https://crater-reports.s3.amazonaws.com/beta-1.60-1/beta-2022-02-22/gh/abesto.byote/log.txt
* https://crater-reports.s3.amazonaws.com/beta-1.60-1/beta-2022-02-22/gh/comcloudway.blueprint/log.txt
* https://crater-reports.s3.amazonaws.com/beta-1.60-1/beta-2022-02-22/gh/hmr.sequoia/log.txt
* https://crater-reports.s3.amazonaws.com/beta-1.60-1/beta-2022-02-22/gh/safesoftware.slpkg/log.txt
* https://crater-reports.s3.amazonaws.com/beta-1.60-1/beta-2022-02-22/gh/sim82.aoc2019/log.txt
* https://crater-reports.s3.amazonaws.com/beta-1.60-1/beta-2022-02-22/gh/vagicc.block_histories/log.txt

```
[INFO] [stderr] thread 'rustc' panicked at 'assertion failed: `(left == right)`
[INFO] [stderr]   left: `7`,
[INFO] [stderr]  right: `4`', compiler/rustc_middle/src/mir/mod.rs:500:13
[INFO] [stderr] stack backtrace:
```

**Comment by bjorn3** [comment]
This seems to be the assertion in https://github.com/rust-lang/rust/blob/0677edc86e342f333d4828b0ee1ef395a4e70fe5/compiler/rustc_middle/src/mir/mod.rs#L512-L523 In other words the `statement_index` is higher than the actual statement count. I'm guessing an optimization removes statements but doesn't update the source infos.

**Comment by Mark-Simulacrum** [comment]
@rustbot ping bisect

Can we try to get a bisection for this failure? Any of the repositories above should work.

**Comment by steffahn** [comment]
Seems to be fixed already

**Comment by steffahn** [comment]
Tested `Jazz` and `slpkg`, it’s fixed for both in latest `beta`; bisections on `slpkg`:

regression in: 9cdefd763b910ffd1d42233a8c752ab5fd84ca4d
fixed in: d4de1f230ca30b7ce08fbf453daebf8b2e7ffcc9

**Comment by Mark-Simulacrum** [comment]
Going to go ahead and close then, thanks @steffahn!
